capsules/extra/src/ieee802154/driver.rs

```diff
@@ -1180,12 +1189,11 @@ impl<'a, M: device::MacDevice<'a>> device::RxClient for RadioDriver<'a, M> {
                 })
                 .unwrap_or(false);
             if read_present {
-                // Encode useful parts of the header in 3 usizes
-                let pans = encode_pans(&header.dst_pan, &header.src_pan);
+                // Encode lqi and useful parts of the header in 3 usizes
                 let dst_addr = encode_address(&header.dst_addr);
                 let src_addr = encode_address(&header.src_addr);
                 kernel_data
-                    .schedule_upcall(upcall::FRAME_RECEIVED, (pans, dst_addr, src_addr))
+                    .schedule_upcall(upcall::FRAME_RECEIVED, (lqi as usize, dst_addr, src_addr))
```

Comment:
I know we didn't promise any stability here, but I'm still concerned about churn. Do you know why pan IDs and src/dst addresses where included in this upcall in the first place? They are in they packet, no? And it's not even possible to fit a 64 bit address in a 32 bit usize.

LQI is not in the packet itself, and that to me makes sense to include in the upcall arguments. I think I would rather see a new upcall be added (eg `FRAME_RECEIVED_LQI` or something) that only provides LQI. This is assuming there is not a use case for including the PAN IDs and src/dst addresses.

I think it might make sense to include other metadata in the future (for example, CRC check status for use cases where even failed crc packets are provided to userspace, or a flag indicating encrypted or not). If we find out that there is no use case for pans/dst_addr/src_addr we can just entirely drop that upcall in the future.